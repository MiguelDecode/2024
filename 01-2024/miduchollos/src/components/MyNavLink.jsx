import { NavLink } from "react-router-dom";

export const MyNavLink = ({ to, children, ...props }) => {
  return (
    <NavLink
      {...props}
      className={({ isActive }) => (isActive ? "is-active" : "")}
      to={to}
    >
      {children}
    </NavLink>
  );
};
