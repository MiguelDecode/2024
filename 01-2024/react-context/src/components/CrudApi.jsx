import { useContext } from "react";
import CrudContext from "../context/CrudContext.jsx";
import CrudForm from "./CrudForm.jsx";
import CrudTable from "./CrudTable.jsx";
import Loader from "./Loader.jsx";
import Message from "./Message.jsx";

const CrudApi = () => {
  const { loading, error, db } = useContext(CrudContext);

  return (
    <div>
      <h2>CRUD API con Context API</h2>
      <article className="grid-1-2">
        <CrudForm />
        {loading && <Loader />}
        {error && (
          <Message
            msg={`Error ${error.status}: ${error.statusText}`}
            bgColor="#dc3545"
          />
        )}
        {db && <CrudTable />}
      </article>
    </div>
  );
};

export default CrudApi;
