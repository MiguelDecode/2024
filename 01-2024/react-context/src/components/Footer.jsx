const Footer = ({ theme, text }) => {
  const { footerTitle } = text;

  return (
    <footer className={theme}>
      <h4>{footerTitle}</h4>
    </footer>
  );
};

export default Footer;
