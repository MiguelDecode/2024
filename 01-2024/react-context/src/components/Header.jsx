const Header = ({
  theme,
  handleTheme,
  handleLanguage,
  text,
  auth,
  handleAuth,
}) => {
  const {
    headerTitle,
    headerSubtitle,
    headerLight,
    headerDark,
    buttonLogin,
    buttonLogout,
  } = text;
  return (
    <header className={theme}>
      <h2>{headerTitle}</h2>
      <h3>{headerSubtitle}</h3>
      <select name="language" onChange={handleLanguage}>
        <option value="es">ES</option>
        <option value="en">EN</option>
      </select>
      <input
        type="radio"
        name="theme"
        id="light"
        onClick={handleTheme}
        value="light"
      />
      <label htmlFor="light">{headerLight}</label>
      <input
        type="radio"
        name="theme"
        id="dark"
        onClick={handleTheme}
        value="dark"
      />
      <label htmlFor="dark">{headerDark}</label>
      <button onClick={handleAuth}>{auth ? buttonLogout : buttonLogin}</button>
    </header>
  );
};

export default Header;
