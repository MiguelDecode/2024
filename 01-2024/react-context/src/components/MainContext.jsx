import { useContext } from "react";
import ThemeContext from "../context/ThemeContext";
import LanguageContext from "../context/LanguageContext";
import AuthContext from "../context/AuthContext";

const MainContext = () => {
  const { theme } = useContext(ThemeContext);
  const { text } = useContext(LanguageContext);
  const { auth } = useContext(AuthContext);


  return (
    <main className={theme}>
      {auth ? <p>{text.mainWelcome}</p> : <p>{text.mainHello}</p>}
      <p>{text.mainContent}</p>
    </main>
  );
};

export default MainContext;
