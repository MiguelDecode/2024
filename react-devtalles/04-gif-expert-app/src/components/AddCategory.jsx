import PropTypes from "prop-types";
import { useState } from "react";

export const AddCategory = ({ addCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (event) => setInputValue(event.target.value);

  const handleSubmit = (event) => {
    event.preventDefault();
    addCategory(inputValue);
    setInputValue("");
  };

  return (
    <form onSubmit={handleSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  addCategory: PropTypes.func.isRequired,
};
