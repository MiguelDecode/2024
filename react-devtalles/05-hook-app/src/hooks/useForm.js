import { useState } from "react";

export const useForm = (initialForm = {}) => {
  const [formState, setFormState] = useState(initialForm);


  const handleInputChange = (event) => {
    const key = event.target.name;
    const value = event.target.value;

    setFormState({ ...formState, [key]: value });
  };

  const handleReset = () => setFormState(initialForm);

  return {
    formState,
    handleInputChange,
    onResetForm: handleReset,
  };
};
