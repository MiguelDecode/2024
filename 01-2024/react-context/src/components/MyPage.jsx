import { useState } from "react";
import Footer from "./Footer";
import Header from "./Header";
import Main from "./Main";

const initialTheme = "light";
const initialLanguage = "es";
const initialAuth = null;

const translations = {
  es: {
    headerTitle: "Mi aplicación sin Context",
    headerSubtittle: "Mi cabecera",
    headerLight: "Claro",
    headerDark: "Oscuro",
    buttonLogin: "Iniciar Sesión",
    buttonLogout: "Cerrar Sesión",
    mainWelcome: "Bienvenid@ Usuario",
    mainHello: "Hola Usuari@",
    mainContent: "Mi contenido principal",
    footerTitle: "Mi pié de página",
  },
  en: {
    headerTitle: "My application without Context",
    headerSubtittle: "My Header",
    headerLight: "Light",
    headerDark: "Dark",
    buttonLogin: "Login",
    buttonLogout: "Logout",
    mainWelcome: "Welcome User",
    mainHello: "Hello User",
    mainContent: "My main content",
    footerTitle: "My footer",
  },
};

const MyPage = () => {
  const [theme, setTheme] = useState(initialTheme);
  const [language, setLanguage] = useState(initialLanguage);
  const [text, setText] = useState(translations[language]);
  const [auth, setAuth] = useState(initialAuth);

  const handleTheme = (event) => setTheme(event.target.value);

  const handleLanguage = (event) => {
    setLanguage(event.target.value);
    setText(translations[event.target.value]);
  };

  const handleAuth = () => {
    if (auth) {
      setAuth(null);
    } else {
      setAuth(true);
    }
  };

  return (
    <div className="my-page">
      <Header
        theme={theme}
        handleTheme={handleTheme}
        handleLanguage={handleLanguage}
        text={text}
        auth={auth}
        handleAuth={handleAuth}
      />
      <Main theme={theme} text={text} auth={auth} />
      <Footer theme={theme} text={text} />
    </div>
  );
};

export default MyPage;
