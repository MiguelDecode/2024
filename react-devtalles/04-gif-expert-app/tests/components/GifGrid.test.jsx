import { render, screen } from "@testing-library/react";
import { GifGrid } from "../../src/components/GifGrid";
import { useFetchGifs } from "../../src/hooks/useFetchGifs";

jest.mock("../../src/hooks/useFetchGifs");

describe("Pruebas en GifGrid", () => {
  const testCategory = "One Punch";

  test("Debe de mostrar el loading inicialmente ", () => {
    useFetchGifs.mockReturnValue({
      images: [],
      isLoading: true,
    });

    render(<GifGrid category={testCategory} />);

    expect(screen.getByText("Cargando..."));
    expect(screen.getByText(testCategory));
  });

  test("Debe de mostrar items cuando se cargan las imÃ¡genes mediante el useFetchGifs", () => {
    const gifs = [
      {
        id: "idTest",
        title: "TestTitle",
        url: "https://test.jpg",
      },
      {
        id: "idTest2",
        title: "TestTitle2",
        url: "https://test2.jpg",
      },
    ];

    useFetchGifs.mockReturnValue({
      images: gifs,
      isLoading: false,
    });

    render(<GifGrid category={testCategory} />);
    expect(screen.getByText("TestTitle"));
    expect(screen.getByText("TestTitle2"));
    expect(screen.getAllByRole("img").length).toBe(2);
  });

  test("Debe de hacer match con el snapshot", () => {
    const { container } = render(<GifGrid category={testCategory} />);

    expect(container).toMatchSnapshot();
  });
});

