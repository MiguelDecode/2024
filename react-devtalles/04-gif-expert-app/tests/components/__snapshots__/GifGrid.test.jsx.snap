// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid Debe de hacer match con el snapshot 1`] = `
<div>
  <h3>
    One Punch
  </h3>
  <div
    class="card-grid"
  >
    <div
      class="card"
    >
      <img
        alt="TestTitle"
        src="https://test.jpg"
      />
      <p>
        TestTitle
      </p>
    </div>
    <div
      class="card"
    >
      <img
        alt="TestTitle2"
        src="https://test2.jpg"
      />
      <p>
        TestTitle2
      </p>
    </div>
  </div>
</div>
`;
